diff --git a/debug/web-tree-sitter.cjs b/debug/web-tree-sitter.cjs
index da0140d03065905b797fb30c772eecb6f0290956..3cdb6692edf56515b2597f3085f73e2949101222 100644
--- a/debug/web-tree-sitter.cjs
+++ b/debug/web-tree-sitter.cjs
@@ -1729,12 +1729,13 @@ async function createModule(options = {}) {
   }, "bytes");
   const getValue = /* @__PURE__ */ __name((ptr, type = "i8") => {
     refresh();
+    ptr >>>= 0;
     switch (type) {
       case "i32":
       case "*":
-        return heap32[ptr >> 2];
+        return heap32[ptr >>> 2];
       case "i16":
-        return heap16[ptr >> 1];
+        return heap16[ptr >>> 1];
       case "i1":
       case "i8":
         return heap8[ptr];
@@ -1750,13 +1751,14 @@ async function createModule(options = {}) {
   }, "getValue");
   const setValue = /* @__PURE__ */ __name((ptr, value, type = "i8") => {
     refresh();
+    ptr >>>= 0;
     switch (type) {
       case "i32":
       case "*":
-        heap32[ptr >> 2] = value;
+        heap32[ptr >>> 2] = value;
         return;
       case "i16":
-        heap16[ptr >> 1] = value;
+        heap16[ptr >>> 1] = value;
         return;
       case "i1":
       case "i8":
diff --git a/debug/web-tree-sitter.js b/debug/web-tree-sitter.js
index 875b2159a4062ef05b3ea2cd68661cb072b8a418..0695d5dd48ebd8ed6c6d0264fc053b13b31aba5b 100644
--- a/debug/web-tree-sitter.js
+++ b/debug/web-tree-sitter.js
@@ -1679,12 +1679,13 @@ async function createModule(options = {}) {
   }, "bytes");
   const getValue = /* @__PURE__ */ __name((ptr, type = "i8") => {
     refresh();
+    ptr >>>= 0;
     switch (type) {
       case "i32":
       case "*":
-        return heap32[ptr >> 2];
+        return heap32[ptr >>> 2];
       case "i16":
-        return heap16[ptr >> 1];
+        return heap16[ptr >>> 1];
       case "i1":
       case "i8":
         return heap8[ptr];
@@ -1700,13 +1701,14 @@ async function createModule(options = {}) {
   }, "getValue");
   const setValue = /* @__PURE__ */ __name((ptr, value, type = "i8") => {
     refresh();
+    ptr >>>= 0;
     switch (type) {
       case "i32":
       case "*":
-        heap32[ptr >> 2] = value;
+        heap32[ptr >>> 2] = value;
         return;
       case "i16":
-        heap16[ptr >> 1] = value;
+        heap16[ptr >>> 1] = value;
         return;
       case "i1":
       case "i8":
diff --git a/web-tree-sitter.cjs b/web-tree-sitter.cjs
index da0140d03065905b797fb30c772eecb6f0290956..3cdb6692edf56515b2597f3085f73e2949101222 100644
--- a/web-tree-sitter.cjs
+++ b/web-tree-sitter.cjs
@@ -1729,12 +1729,13 @@ async function createModule(options = {}) {
   }, "bytes");
   const getValue = /* @__PURE__ */ __name((ptr, type = "i8") => {
     refresh();
+    ptr >>>= 0;
     switch (type) {
       case "i32":
       case "*":
-        return heap32[ptr >> 2];
+        return heap32[ptr >>> 2];
       case "i16":
-        return heap16[ptr >> 1];
+        return heap16[ptr >>> 1];
       case "i1":
       case "i8":
         return heap8[ptr];
@@ -1750,13 +1751,14 @@ async function createModule(options = {}) {
   }, "getValue");
   const setValue = /* @__PURE__ */ __name((ptr, value, type = "i8") => {
     refresh();
+    ptr >>>= 0;
     switch (type) {
       case "i32":
       case "*":
-        heap32[ptr >> 2] = value;
+        heap32[ptr >>> 2] = value;
         return;
       case "i16":
-        heap16[ptr >> 1] = value;
+        heap16[ptr >>> 1] = value;
         return;
       case "i1":
       case "i8":
diff --git a/web-tree-sitter.js b/web-tree-sitter.js
index 875b2159a4062ef05b3ea2cd68661cb072b8a418..0695d5dd48ebd8ed6c6d0264fc053b13b31aba5b 100644
--- a/web-tree-sitter.js
+++ b/web-tree-sitter.js
@@ -1679,12 +1679,13 @@ async function createModule(options = {}) {
   }, "bytes");
   const getValue = /* @__PURE__ */ __name((ptr, type = "i8") => {
     refresh();
+    ptr >>>= 0;
     switch (type) {
       case "i32":
       case "*":
-        return heap32[ptr >> 2];
+        return heap32[ptr >>> 2];
       case "i16":
-        return heap16[ptr >> 1];
+        return heap16[ptr >>> 1];
       case "i1":
       case "i8":
         return heap8[ptr];
@@ -1700,13 +1701,14 @@ async function createModule(options = {}) {
   }, "getValue");
   const setValue = /* @__PURE__ */ __name((ptr, value, type = "i8") => {
     refresh();
+    ptr >>>= 0;
     switch (type) {
       case "i32":
       case "*":
-        heap32[ptr >> 2] = value;
+        heap32[ptr >>> 2] = value;
         return;
       case "i16":
-        heap16[ptr >> 1] = value;
+        heap16[ptr >>> 1] = value;
         return;
       case "i1":
       case "i8":
